src/common/low_precision_transformations/src/strided_slice.cpp

```diff
@@ -18,108 +18,133 @@ namespace low_precision {
+        }
     }
 
-    const auto stridedSlice = ov::as_type_ptr<ngraph::opset1::StridedSlice>(strSlice);
-
-    auto beginMask = stridedSlice->get_begin_mask();
-    auto endMask = stridedSlice->get_end_mask();
-    for (size_t i = 0; i < constantShape.size(); ++i) {
-        if ((beginMask.size() <= i) && (endMask.size() <= i)) {
-            break;
-        }
+    // step #2: update original begin & end & strides
+    const auto strided_slice = ov::as_type_ptr<ov::op::v1::StridedSlice>(node);
+    auto begin = ov::as_type_ptr<ov::op::v0::Constant>(strided_slice->get_input_node_shared_ptr(1))->cast_vector<int64_t>();
+    auto end = ov::as_type_ptr<ov::op::v0::Constant>(strided_slice->get_input_node_shared_ptr(2))->cast_vector<int64_t>();
+    auto strides = ov::as_type_ptr<ov::op::v0::Constant>(strided_slice->get_input_node_shared_ptr(3))->cast_vector<int64_t>();
+    auto begin_mask = strided_slice->get_begin_mask();
+    auto end_mask = strided_slice->get_end_mask();
+    for (auto i = 0ull; i < new_constant_shape.size(); ++i) {
         // don't slice constant if current dimension is 1
-        if (constantShape[i] == 1ul) {
-            beginMask[i] = 1ul;
-            endMask[i] = 1ul;
+        if (new_constant_shape[i] == 1ull) {
+            if (i < begin.size()) {
+                begin[i] = 0;
+            }
+            if (i < end.size()) {
+                end[i] = 1;
+            }
+
+            if (i < strides.size()) {
+                strides[i] = 1;
+            }
+
+            if (i < begin_mask.size()) {
+                begin_mask[i] = 1;
+            }
+
+            if (i < end_mask.size()) {
+                end_mask[i] = 1;
+            }
         }
     }
 
-    const auto result = fold<ngraph::opset1::StridedSlice>(
-        constant,
-        stridedSlice->input_value(1),
-        stridedSlice->input_value(2),
-        stridedSlice->input_value(3),
-        beginMask,
-        endMask,
-        stridedSlice->get_new_axis_mask(),
-        stridedSlice->get_shrink_axis_mask(),
-        stridedSlice->get_ellipsis_mask());
-
-    return ov::as_type_ptr<opset1::Constant>(NetworkHelper::toScalarIfPossible(result));
+    // step #3: final step: dequantizatin constant folding
+    const auto result = fold<ov::op::v1::StridedSlice>(
+        new_constant,
+        std::make_shared<opset1::Constant>(element::i64, Shape{ begin.size() }, begin),
+        std::make_shared<opset1::Constant>(element::i64, Shape{ end.size() }, end),
+        std::make_shared<opset1::Constant>(element::i64, Shape{ strides.size() }, strides),
+        begin_mask,
+        end_mask,
+        strided_slice->get_new_axis_mask(),
+        strided_slice->get_shrink_axis_mask(),
+        strided_slice->get_ellipsis_mask());
+
+    new_constant = ov::as_type_ptr<opset1::Constant>(NetworkHelper::toScalarIfPossible(result));
+
+    if (shape_size(new_constant->get_shape()) == 1ul) {
+        return NetworkHelper::toScalar(new_constant);
+    }
```

Comment:
Looks like we don't need this code as we call `NetworkHelper::toScalarIfPossible` at the previous line

Reply:
done